libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -6,9 +6,14 @@
   implement a simple ringbuffer of bytes
  */
 
+static inline uint32_t align_power2(uint32_t u)
+{
+    return 1U << ((sizeof(u) * 8) - __builtin_clz(u - 1));
+}
+
 ByteBuffer::ByteBuffer(uint32_t _size)
 {
-    size = _size;
+    size = align_power2(_size);
```

Comment:
The first time I took a look in the code I also thought that was too much effort (and possible memory waste) just to get rid of a "%". So +1 to not do that.